jetbot/bot.py

```diff
@@ -451,48 +470,68 @@ async def channels(ctx: commands.Context):
+                            await channel.edit(slowmode_delay=5, sync_permissions=True)
+
+                            # Handle channel type first
+                            if "type" in channel_config:
+                                if channel_config["type"] == "forum" and not isinstance(channel, ForumChannel):
+                                    await channel.edit(type=ChannelType.forum)
+                                elif channel_config["type"] == "stage_voice" and not isinstance(channel, StageChannel):
+                                    await channel.edit(type=ChannelType.stage_voice)
+
+                    channels_to_update = await self.product_channels(item)
+
+                    for channel in channels_to_update:
                         product_emoji = await self.product_emoji(item["emoji_name"])
-                        title = "{0} {1} {0}".format(product_emoji, item["name"]) if product_emoji else item["name"]
-                        if channel.topic != title:
-                            await channel.edit(topic=title)
-
-                        # Set permissions
-                        await channel.set_permissions(guild.default_role,
-                                                    send_messages=False,
-                                                    send_messages_in_threads=False,
-                                                    create_private_threads=False,
-                                                    create_public_threads=False)
-                    else:
-                        # Set the topic
                         product_category = category.name.lower().strip()
-                        product_emoji = await self.product_emoji(item["emoji_name"])
-                        title = title_map[product_category] if product_category in title_map else default_title
-                        title = title.format(item["name"])
+                        channel_description = None
+
+                        if "channels" in item:
+                            for ch_config in item["channels"]:
+                                if ch_config["name"] == channel.name:
+                                    channel_description = ch_config["description"]
+                                    break
+
+                        if channel_description:
+                            title = channel_description
+                        else:
+                            title = title_map[product_category] if product_category in title_map else default_title
+                            title = title.format(item["name"])
                         title = "{0} {1} {0}".format(product_emoji, title) if product_emoji else title
+
                         if channel.topic != title:
                             await channel.edit(topic=title)
 
+                        # Handle permissions
+                        for ch_config in item["channels"]:
```

Comment:
Ditto again here, we're having to do a bunch of extra looping which doesn't make sense